src/mlpack/core/data/csv_parser_impl.hpp

```diff
@@ -0,0 +1,250 @@
+
+        return true;
+      }
+    }
+
+    char* endptr = nullptr;
+
+    if (std::is_floating_point<typename MatType::elem_type>::value || std::is_integral<typename MatType::elem_type>::value)
+    {
+      val = typename MatType::elem_type(std::strtod(str, &endptr));
+    }
+    else
+    {
+      if (std::is_signed<typename MatType::elem_type>::value)
+      {
+        val = typename MatType::elem_type(std::strtoll(str, &endptr, 10));
+      }
+      else
+      {
+        if (str[0] == '-')
+        {
+          val = typename MatType::elem_type(0);
+          return true;
+        }
+
+        val = typename MatType::elem_type( std::strtoull(str, &endptr, 10) );
+      }      
+    }
+
+    if (str == endptr)
+    {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Loads the data from the csv file
+   * into the give MatType
+   */
+  template<typename MatType>
+  bool LoadCSVV(MatType& x, std::fstream& f, std::string&)
+  {
+    bool load_okay = f.good();
+
+    f.clear();
+    const std::fstream::pos_type pos1 = f.tellg();
+
+    size_t f_n_rows = 0;
+    size_t f_n_cols = 0;
+
+    std::string line_string;
+    std::stringstream line_stream;
+
+    std::string token;
+
+    while(f.good() && load_okay)
+    {
+      std::getline(f, line_string);
+
+      if (line_string.size() == 0)
+      { 
+        break; 
+      }
+
+      line_stream.clear();
+      line_stream.str(line_string);
+
+      size_t line_n_cols = 0;
+
+      while(line_stream.good())
+      {
+        // reading each element of the row
+        std::getline(line_stream, token, ',');
+        ++line_n_cols;
+      }
+
+      if (f_n_cols < line_n_cols)
+      {
+        f_n_cols = line_n_cols;
+      }
+      
+      ++f_n_rows;
+    }
+
+    f.clear();
+    f.seekg(pos1);
+
+    x.zeros(f_n_rows, f_n_cols);
+
+    size_t row = 0;
+
+    while (f.good())
+    {
+      std::getline(f, line_string);
+
+      if (line_string.size() == 0)
+      {
+        break;
+      }
+
+      line_stream.clear();
+      line_stream.str(line_string);
+
+      size_t col = 0;
+
+      while (line_stream.good())
+      {
+        std::getline(line_stream, token, ',');
+
+        ConvertToken<MatType>(x.at(row, col), token);
+
+        ++col;
+      }
+      ++row;
+    }
+
+    return load_okay;		
+  }
+
+  template <typename MatType>
+  bool LoadData(const std::string& name, MatType& x, const mlpack::data::file_type type)
+  {
+    // bool load_okay = false;
+    std::string err_msg;
+    std::string print_status;
+    std::fstream f;
+
+    f.open(name.c_str(), std::fstream::in);
+
+    switch (type)
+    {
+      case mlpack::data::file_type::csv_ascii:
+        return LoadCSVV<MatType>(x, f, print_status);
+
+      case mlpack::data::file_type::file_type_unknown:
+        return true;
+
+      case mlpack::data::file_type::auto_detect:
+        return true;
+
+      case mlpack::data::file_type::raw_ascii:
```

Comment:
@heisenbuug I do not see what is the objective of this function, should it replace `Load` function declared in line 89 in `load_impl.hpp`? or are you planning to call it from there?

Reply:
Yea, I think we can remove this function, and if at all we need any modifications I can do them in `load_impl.hpp`...